crates/sui-types/src/attestation.rs

```diff
@@ -0,0 +1,539 @@
+        seq.serialize_element(&self.unprotected)?;
+        seq.serialize_element(&self.payload)?;
+        seq.serialize_element(&self.signature)?;
+        seq.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for CoseSign1 {
+    fn deserialize<D>(deserializer: D) -> Result<CoseSign1, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        use serde::de::{Error, SeqAccess, Visitor};
+        use std::fmt;
+
+        struct CoseSign1Visitor;
+
+        impl<'de> Visitor<'de> for CoseSign1Visitor {
+            type Value = CoseSign1;
+
+            fn expecting(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                f.write_str("a possibly tagged CoseSign1 structure")
+            }
+
+            fn visit_seq<A>(self, mut seq: A) -> Result<CoseSign1, A::Error>
+            where
+                A: SeqAccess<'de>,
+            {
+                // This is the untagged version
+                let protected = match seq.next_element()? {
+                    Some(v) => v,
+                    None => return Err(A::Error::missing_field("protected")),
+                };
+
+                let unprotected = match seq.next_element()? {
+                    Some(v) => v,
+                    None => return Err(A::Error::missing_field("unprotected")),
+                };
+                let payload = match seq.next_element()? {
+                    Some(v) => v,
+                    None => return Err(A::Error::missing_field("payload")),
+                };
+                let signature = match seq.next_element()? {
+                    Some(v) => v,
+                    None => return Err(A::Error::missing_field("signature")),
+                };
+
+                Ok(CoseSign1 {
+                    protected,
+                    unprotected,
+                    payload,
+                    signature,
+                })
+            }
+
+            fn visit_newtype_struct<D>(self, deserializer: D) -> Result<CoseSign1, D::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                // This is the tagged version: we ignore the tag part, and just go into it
+                deserializer.deserialize_seq(CoseSign1Visitor)
+            }
+        }
+
+        deserializer.deserialize_any(CoseSign1Visitor)
+    }
+}
+
+impl CoseSign1 {
+    /// Parse CBOR bytes into struct. Adapted from https://github.com/awslabs/aws-nitro-enclaves-cose/blob/main/src/sign.rs
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, NitroError> {
+        let tagged_value: ciborium::value::Value = ciborium::de::from_reader(bytes)
+            .map_err(|e| NitroError::InvalidCoseSign1(e.to_string()))?;
+
+        let (tag, value) = match tagged_value {
+            ciborium::value::Value::Tag(tag, box_value) => (Some(tag), *box_value),
+            other => (None, other),
+        };
+
+        // Validate tag (18 is the COSE_Sign1 tag)
+        match tag {
+            None | Some(18) => (),
```

Comment:
it seems that in the official aws doc it can be none: `Accepts untagged structures or structures with tag 18.` im not sure why

https://github.com/awslabs/aws-nitro-enclaves-cose/blob/main/src/sign.rs#L324